src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1098,28 +1253,56 @@ bool is_object_properties_superset(
             auto regex = re2::RE2(as_string_view(propPattern));
             if (re2::RE2::PartialMatch(pname, regex)) {
                 pattern_match_found = true;
-                if (!is_superset(ctx, schemaPattern, schema)) {
-                    // not compatible
-                    return false;
+
+                auto prop_res = is_superset(
+                  ctx, schemaPattern, schema, prop_path);
+
+                if (prop_res.has_error()) {
+                    res.merge(std::move(prop_res));
+                    res.emplace<json_incompatibility>(
+                      std::move(prop_path),
+                      json_incompatibility_type::
+                        property_removed_not_covered_by_partially_open_content_model);
                 }
+                break;
```

Comment:
Is this in the right scope? I think it should be either with the error, or after the loop if there is any error, depending on whether all errors should be collected.

Reply:
Good spot! I've fixed this now and added a test for it. It looks like the pre-PR implementation was correct and it's enough to report the first error, so I've moved the `break` to be with the error.